pkgs/by-name/mv/mvfst/package.nix

```diff
@@ -0,0 +1,149 @@
+{
+  lib,
+  stdenv,
+
+  fetchFromGitHub,
+
+  cmake,
+  ninja,
+  removeReferencesTo,
+
+  folly,
+  gflags,
+  glog,
+  apple-sdk_11,
+  darwinMinVersionHook,
+
+  fizz,
+
+  gtest,
+
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mvfst";
+  version = "2024.11.11.00";
+
+  outputs = [
+    "bin"
+    "out"
+    "dev"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "facebook";
+    repo = "mvfst";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-AQNzVe0z6psj/e4kEtjs6aHVRjqJhvkmd16XiXExmwk=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    removeReferencesTo
+  ];
+
+  buildInputs =
+    [
+      folly
+      gflags
+      glog
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      apple-sdk_11
+      (darwinMinVersionHook "11.0")
+    ];
+
+  propagatedBuildInputs = [
+    fizz
+  ];
+
+  checkInputs = [
+    gtest
+  ];
+
+  cmakeFlags =
+    [
+      (lib.cmakeBool "BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))
+
+      (lib.cmakeBool "CMAKE_INSTALL_RPATH_USE_LINK_PATH" true)
+
+      (lib.cmakeBool "BUILD_TESTS" finalAttrs.finalPackage.doCheck)
+
+      (lib.cmakeFeature "CMAKE_INSTALL_PREFIX" (placeholder "dev"))
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      # Homebrew sets this, and the shared library build fails without
+      # it. I don‘t know, either. It scares me.
+      (lib.cmakeFeature "CMAKE_SHARED_LINKER_FLAGS" "-Wl,-undefined,dynamic_lookup")
```

Comment:
this is on x86 darwin only? mvfst built for me on aarch64 tho don't recall if I was at the latest rev, maybe late October.

Reply:
I think a combination of updating + building shared libraries breaks it. Homebrew sets it unconditionally: https://github.com/Homebrew/homebrew-core/commit/7307d8bf53e9c4100e9f52bca6191fb6f660aff8.

Reply:
I’ve since learned that although this option is deprecated, it doesn’t force a flat namespace, and it basically just makes `ld64` behave more like the GNU linker with regards to undefined symbols. An upstream fix to add the necessary libraries would be ideal, but the final executables are at least guaranteed to not have anything unbound.